src/gui/appearance/LookAndFeel.cpp

```diff
@@ -48,8 +48,8 @@ const juce::Typeface::Ptr LookAndFeel::getKarlaTypeface()
 int LookAndFeel::getTabButtonBestWidth(juce::TabBarButton& button,
                                        int /*tabDepth*/)
 {
-    float totalWidth = button.getTabbedButtonBar().getWidth();
-    float numTabs = button.getTabbedButtonBar().getNumTabs();
+    float totalWidth = static_cast<float>(button.getTabbedButtonBar().getWidth());
+    float numTabs = static_cast<float>(button.getTabbedButtonBar().getNumTabs());
```

Comment:
Very small thing but cases where there is something like `T myVariable = static_cast<T>(somethingElse);`, I suggest using `auto` to declare the variable type. So have `auto myVariable = static_cast<T>(somethingElse);`. For more info see [LLVM "Use Auto Type Deduction to Make Code More Readable"](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable) and [`clang-tidy` modernize-use-auto rule](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-auto.html#cast-expressions).

This will be picked up by Trunk so might as well change it so the checks don't fail when I run them before merge!

Reply:
> Very small thing but cases where there is something like `T myVariable = static_cast<T>(somethingElse);`, I suggest using `auto` to declare the variable type. So have `auto myVariable = static_cast<T>(somethingElse);`. For more info see [LLVM "Use Auto Type Deduction to Make Code More Readable"](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable) and [`clang-tidy` modernize-use-auto rule](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-auto.html#cast-expressions).
> 
> This will be picked up by Trunk so might as well change it so the checks don't fail when I run them before merge!

Thanks for this document very helpful for learning more! 
A question that came to mind is what are the differences between these [LLVM "Use Auto Type Deduction to Make Code More Readable"](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable) and https://github.com/cpplint/cpplint is just that the former has rules/coding standards specifically picked up by the clang LLVM based compiler or are there any other differences as well?

Reply:
If you think there's a compelling reason to also check things with `cpplint` because `clang-tidy` doesn't cover it, since we're using Trunk (and hopefully will soon have CI integrated) we can easily add that.

Reply:
just did some research to better understand this so there's no need for `cpplint` as its just a style checker rather than a static analysis tool